src/cljc/matcher_combinators/dispatch.cljc

```diff
@@ -1,4 +1,6 @@
 (ns matcher-combinators.dispatch
+  "core/Matchers$match gets bound to dispatch functions defined in this namespace
+  so that we can override them at runtime."
```

Comment:
:+1: 

> This namespace provides a level of indirection between the matcher assigned to a datatype and the matcher implementation. This indirection allows for redefinition at runtime, necessary for the `match-with` feature

Is how I would write it, not sure if any of that seems useful to incorporate

Reply:
I combined ideas from this with what was  there before.